### PR TITLE
1500 actions sendinblue

### PR DESCRIPTION
**Pull request by alysonturing**
### FYI:
On the original ticket was request to create triggers for (**New campaign status, New contact added to a list, New or updated contact**), I couldn't find in the docs exactly how to track these parameters, so I implemented all webhook events;

### Implemented Triggers:
**For type Transaction:**
sent, request, delivered, hardBounce, softBounce, blocked, spam, invalid, deferred, click, opened, uniqueOpened and unsubscribed
**For type Marketing:** 
spam, opened, click, hardBounce, softBounce, unsubscribed, listAddition, delivered

### Implemented Actions:
- [x] Add or update contact
- [x] Send transactional email

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible
- [ ] [Create subscription webhooks sources (vs polling sources) whenever possible](https://pipedream.com/docs/components/guidelines/#webhook-vs-polling-sources)
- [ ] [Always emit a `summary` with events](https://pipedream.com/docs/components/guidelines/#emit-a-summary)
- [ ] [Use a built-in deduping strategy wherever possible](https://pipedream.com/docs/components/guidelines/#deduping)
- [ ] When building webhook sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#webhook-sources)
- [ ] When building polling sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#polling-sources)

File: components/sendinblue/actions/add-or-update-contact/add-or-update-contact.mjs [hunk 1]
```diff
@@ -0,0 +1,101 @@
+import sendinBlueApp from "../../sendinblue.app.mjs";
+
+// legacy_hash_id: a_0Mi7n5
+export default {
+  key: "sendinblue-add-or-update-contact",
+  name: "Add or Updated a contact",
+  description: "Add or Updated a contact and list",
+  version: "0.0.79",
```
Comment (dylburger) [inline components/sendinblue/actions/add-or-update-contact/add-or-update-contact.mjs:8]: You'll want to update these to read `0.0.1`. See https://pipedream.com/docs/components/guidelines/#versioning

File: components/sendinblue/actions/add-or-update-contact/add-or-update-contact.mjs [hunk 2]
```diff
@@ -0,0 +1,101 @@
+import sendinBlueApp from "../../sendinblue.app.mjs";
+
+// legacy_hash_id: a_0Mi7n5
```
Comment (dylburger) [inline components/sendinblue/actions/add-or-update-contact/add-or-update-contact.mjs:3]: This was a special comment we used when we converted a set of actions from our previous version to the latest component action API, but you won't need to include this in any new components.

File: components/sendinblue/actions/add-or-update-contact/add-or-update-contact.mjs [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+export default {
+  key: "sendinblue-add-or-update-contact",
+  name: "Add or Updated a contact",
```
Comment (dylburger) [inline components/sendinblue/actions/add-or-update-contact/add-or-update-contact.mjs:6]: "Add or Update"

File: components/sendinblue/actions/add-or-update-contact/add-or-update-contact.mjs [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  description: "Add or Updated a contact and list",
```
Comment (dylburger) [inline components/sendinblue/actions/add-or-update-contact/add-or-update-contact.mjs:7]: "Add or Update"

Also, is the "and list" intentional? The key and name just say add or update contact, so the "list" stood out.

**Review by vellames-turing — APPROVED** [review]
Looks great for me!

**Comment by DilanAthukorala** [comment]
@alysonturing When we are receiving events it's difficult identify what type of event it is without clicking on it because there's no information in the event list other than the received time. Can we have some more info like the event type listed there?

image

**Comment by alysonturing** [comment]
Yes it's possible, should I send a new PR or just update this one?

**Comment by vellames-turing** [comment]
Update this one so I can review your changes again

**Review by vellames-turing — APPROVED** [review]

**Comment by DilanAthukorala** [comment]
Verification completed. 
FYI: Unable to test marketing webhooks since it requires a paid sendinblue account

**Pull request merged**